tests/app/IntegrationTests/UI.IntegrationTests/CommandsDialogs/FormCommitTests.cs

```diff
@@ -111,15 +111,17 @@ public void Should_not_update_committer_info_on_form_activated()
         [Test]
         public void Should_display_branch_and_no_remote_info_in_statusbar()
         {
-            _referenceRepository.CheckoutBranch("master");
+            string defaultBranchName = GitConfigurationHelper.GetSetting("init.defaultbranch", "master");
```

Comment:
🤔
```suggestion
            string defaultBranchName = _remoteReferenceRepository.Module.GetEffectiveSetting("init.defaultbranch", "master");
```
Perhaps, as extension method for `IGitModule` even, e.g.
```
static string GetDefaultBranchName(this IGitModule module) => module.GetEffectiveSetting("init.defaultbranch", "master");
```
Then it is just:
```suggestion
            string defaultBranchName = _remoteReferenceRepository.Module.GetDefaultBranchName();
```

Reply:
Oh, that's much better because it takes into account whether the setting has been given a unique value in _this particular_ repository. :-)

Reply:
Hmm, there's actually a gap. If you `git init --initial-branch=foobar`, then your repo still doesn't have any refs in it, but the first commit you do will be given the ref name `foobar` -- but `init.defaultbranch` isn't changed, even in the context of the repository. Looking into how to tell what the _actual initial branch name for the current repository_ is...

Reply:
Looks like, barring corruption due to e.g. interrupted operations, the initial state is when `HEAD` contains a local head ref but that ref isn't actually defined.

Reply:
A wrinkle on that: Looks like `ReferenceRepository`, the way it is currently used, _always_ immediately creates an initial commit. So, any logic based on HEAD being set to a non-existing ref is out of the window.

But then, before each test that shares a `ReferenceRepository` instance, `ResetRepo` is called, but it doesn't actually "reset" the repo, except in the technical sense of running `git reset --hard`. It does a few specific actions, but if the repo has additional branches created, they aren't removed, and if the repo's _current_ branch is changed, it isn't returned to its initial setting. In testing, in between tests, I poked around into the underlying storage for the `ReferenceRepository` being used by, in this case, `BlameControlTests`, and I ran `git checkout -b splonk`, and the next test did all of its work inside the `splonk` branch. 😛

<img width="1511" height="905" alt="image" src="https://github.com/user-attachments/assets/2c5ed04a-b2a2-4425-a72d-6580ad880fad" />

In practice, the tests pass, but the tests _don't actually know,_ for certain, the state of the repository they're about to work with. If a test made a mess in the `ReferenceRepository`, subsequent tests would have unpredictable results.

Right now, if the test after I ran `git checkout -b splonk` had been one of the ones that depends on the default branch name, the test would have failed, because not only does it need to know whether the default branch is `master` or `main` or whatnot, _it also assumes that HEAD is pointing at that branch_.

It would be less efficient, but lead to more repeatable testing, if `ReferenceRepository` instances were not shared between tests, and _every_ test initialized a fresh, blank repository to work in, guaranteed known starting point. On the surface, this raises the question of performance impacts, but `git init` isn't being added in a vacuum, it's taking the place of what currently runs in `ReferenceRepository.Reset`:

- `git reset --hard`
- `RemoveUntrackedFiles()`
- `Network.Remotes.Remove(remoteName)` for each remote
- `Config.Set` twice

Also, I just instrumented it and did a full test run, and `ReferenceRepository.ResetRepo` was called a total of 316 times over a 10.8 minute test run, or an average of once every two seconds. It's not a huge number. If I alter `ResetRepo` to _just always make a new repository_, then the same test run comes out to 12.3 minutes -- measurable but manageable? Acceptable for the sake of good test isolation?

Thoughts?